### PR TITLE
Add sensor referenceFrame/frequency TypeError tests

### PR DESCRIPTION
**Pull request by xiuqijix**

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None


<h2>Failing Jobs</h2>
<ul>

<li>firefox:nightly</li>

</ul>



<h2>Unstable Results</h2>
  
  
  
  <h3>Browser: "Firefox Nightly"<small></small></h3>
  <p>View in: <a href="http://pulls.web-platform-tests.org/job/25920.4">WPT PR Status</a> |
      <a href="https://travis-ci.org/w3c/web-platform-tests/jobs/353239684">TravisCI</a></p>
  <table>
    <tr>
      <th>Test</th>
      <th>Subtest</th>
      <th>Results</th>
      <th>Messages</th>
    </tr>
  
  
  
  
  <tr>
    <td><code>/accelerometer/Accelerometer.https.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Accelerometer Test</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>ReferenceError: Accelerometer is not defined</code><br /></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/accelerometer/Accelerometer_insecure_context.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Accelerometer is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>GravitySensor is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>LinearAccelerationSensor is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/ambient-light/AmbientLightSensor.https.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>AmbientLightSensor Test</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>ReferenceError: AmbientLightSensor is not defined</code><br /></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/ambient-light/AmbientLightSensor_insecure_context.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>AmbientLightSensor is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/geolocation-sensor/GeolocationSensor.https.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>GeolocationSensor Test</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>ReferenceError: GeolocationSensor is not defined</code><br /></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/geolocation-sensor/GeolocationSensor_insecure_context.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>GeolocationSensor is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/gyroscope/Gyroscope.https.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Gyroscope Test</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>ReferenceError: Gyroscope is not defined</code><br /></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/gyroscope/Gyroscope_insecure_context.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Gyroscope is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/magnetometer/Magnetometer.https.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Magnetometer Test</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>ReferenceError: Magnetometer is not defined</code><br /></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/magnetometer/Magnetometer_insecure_context.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Magnetometer is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>UncalibratedMagnetometer is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/orientation-sensor/OrientationSensor.https.html</code></td>
    <td>&nbsp;</td>
    <td>ERROR: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Test AbsoluteOrientationSensor.populateMatrix() method works correctly.</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>AbsoluteOrientationSensor is not defined</code><br /></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Test AbsoluteOrientationSensor.quaternion return a four-element FrozenArray.</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>AbsoluteOrientationSensor is not defined</code><br /></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Test RelativeOrientationSensor.populateMatrix() method works correctly.</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>RelativeOrientationSensor is not defined</code><br /></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>Test RelativeOrientationSensor.quaternion return a four-element FrozenArray.</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>RelativeOrientationSensor is not defined</code><br /></td>
  </tr>
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/orientation-sensor/OrientationSensor_insecure_context.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>AbsoluteOrientationSensor is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>RelativeOrientationSensor is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/proximity/ProximitySensor.https.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>ProximitySensor Test</code></td>
    <td>FAIL: 6<br /></td>
    <td><code>ReferenceError: ProximitySensor is not defined</code><br /></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/proximity/ProximitySensor_insecure_context.html</code></td>
    <td>&nbsp;</td>
    <td>OK: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>ProximitySensor is not exposed in an insecure context</code></td>
    <td>PASS: 6<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  </table>

File: sensors/generic-sensor-tests.js [hunk 1]
```diff
@@ -259,6 +259,22 @@ function runGenericSensorTests(sensorType) {
...
+      -Infinity,
+      {},
+      undefined
+    ];
+    invalidFreqs.map(freq => {
+      assert_throws(new TypeError(),
+                    () => { new sensorType({frequency: freq}) },
+                    `when freq is ${freq}`);
+    });
+  }, `${sensorType.name}: throw 'TypeError' if frequency is an incorrect value type`);
```
Comment (Honry) [inline sensors/generic-sensor-tests.js]: The test name is not quite appropriate, as `Infinity` is also typeof `number`.  So I think we can just describe it as "throw 'TypeError' if frequency is invalid"

Comment (kaixinjxq) [reply sensors/generic-sensor-tests.js]: Thx @Honry, updated.

**Review by Honry — APPROVED** [review]
Thanks, LGTM

**Pull request merged**